pkg/endpoint/policy.go

```diff
@@ -109,7 +109,14 @@ func (e *Endpoint) removeOldFilter(owner Owner, labelsMap *identityPkg.IdentityC
 	port := uint16(filter.Port)
 	proto := uint8(filter.U8Proto)
 
-	for _, sel := range filter.FromEndpoints {
+	fromEndpointsSelectors := filter.FromEndpoints
+	if fromEndpointsSelectors == nil {
+		fromEndpointsSelectors = []api.EndpointSelector{
+			api.NewWildcardEndpointSelector(),
+		}
+	}
+
+	for _, sel := range fromEndpointsSelectors {
```

Comment:
Out of curiosity, does this means all SecIDs will be selected?

Reply:
For the case where the endpoint selector is empty, the new code inserts a wildcard selector which should select all SecIDs, yes.